app/src/main/java/com/sun/mygif/ui/favorite/FavoriteAdapter.kt

```diff
@@ -0,0 +1,68 @@
+package com.sun.mygif.ui.favorite
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.sun.mygif.R
+import com.sun.mygif.data.model.GifFavorite
+import com.sun.mygif.ui.base.BaseRecyclerViewAdapter
+import com.sun.mygif.ui.base.BaseViewHolder
+import com.sun.mygif.utils.screenWidth
+import com.sun.mygif.widget.BASE_WIDTH
+import kotlinx.android.synthetic.main.item_favorite.view.*
+
+class FavoriteAdapter(
+    private val onItemClick: (gifFavorite: GifFavorite) -> Unit,
+    private val onItemDelete: (position: Int, gifFavorite: GifFavorite) -> Unit
+) : BaseRecyclerViewAdapter<GifFavorite, FavoriteAdapter.FavoriteViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, position: Int): FavoriteViewHolder = FavoriteViewHolder(
+        LayoutInflater.from(parent.context).inflate(R.layout.item_favorite, parent, false),
+        onItemClick,
+        onItemDelete
+    )
+
+    override fun onBindViewHolder(viewHolder: FavoriteViewHolder, position: Int) =
+        viewHolder.onBindDataPosition(position)
+
+    inner class FavoriteViewHolder(
+        itemView: View,
+        private val onItemClick: (gifFavorite: GifFavorite) -> Unit,
+        private val onItemDelete: (position: Int, gifFavorite: GifFavorite) -> Unit
+    ) : BaseViewHolder<GifFavorite>(itemView) {
+
+        init {
+            itemView.apply {
+                gifItemFavorite.setOnClickListener {
+                    item?.let { onItemClick(it) }
+                }
+                iconDeleteItemFavorite.setOnClickListener {
+                    item?.let { onItemDelete(positionItem, it) }
+                }
+                titleItemFavorite?.run {
+                    width = screenWidth ushr 1
+                    isSelected = true
+                }
+            }
+        }
+
+        override fun onBindDataPosition(position: Int) {
+            super.onBindDataPosition(position)
+            item = getItem(position)
+
+            item?.let {
+                with(itemView) {
+                    gifItemFavorite?.run {
+                        setRatioSize(
+                            ratio = it.gif.width.toFloat() / it.gif.height,
```

Comment:
it đây là gì ? khi sử dụng nhiều it lồng nhau thì e nên định nghĩa lại cho người đọc dễ hiểu